### ISSUE TITLE
How can I convert Dictionary<String,Any> to JSON?

### ISSUE THREAD

**Issue by omidgolparvar** [saoudrizwan/DynamicJSON#2]
I used the source code in a playground, and when I want to create a JSON object from a dictionary (it is [String: Any]), it gives me this:

```error: Execution was interrupted, reason: EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0).```

I used this:

```swift
let dictionary: [String: Any] = [
  "name"    : "Omid",
  "age"     : 29,
  "married" : false
]

let dictionaryAsJSON = JSON(dictionary)

print(dictionaryAsJSON.description)
```

I think it is only in runtime, because Xcode does not show any issue, until running plaground.

<img width="520" alt="screen shot 2019-01-03 at 19 36 06" src="https://user-images.githubusercontent.com/5128342/50647861-eef02900-0f8e-11e9-9505-77a6b1570ca3.png">

**Comment by saoudrizwan** [comment]
It looks like this is a **bug in Playgrounds** where initializing an enum with an associated value causes the Playground to be uncertain of how much memory the object is taking up, causing the sig crash. For example, even if you were to put in something like
```swift
let val = JSON.string("adf")
```
it would crash as well.
I will investigate this further and report back, but your example works perfectly fine in a normal project. Here's a shorter version of your code:
```swift
let dictionary: JSON = [
	"name"    : "Omid",
	"age"     : 29,
	"married" : false
]

print(dictionary)
```

**Comment by saoudrizwan** [comment]
## Update
I've narrowed down the problem to the way `JSON` conforms to `Swift.CustomStringConvertible, Swift.CustomDebugStringConvertible`. It seems like Playgrounds calls the implicit `debugDescription` property on its own (to display it on the debug panel on the right), so as soon as it does that, it calls my `debugDescription` override which uses `JSONSerialization.jsonObject(with: data(options: .prettyPrinted), options: [])` to convert to a pretty printed JSON object.
Interestingly, it seems that using `self` in a `CustomDebugStringConvertible` implementation causes infinite recursion in the logger. Since the memory allocated to printing is probably limited, and this method is memory intensive, we get this crash.
A similar bug was filed regarding this issue: https://bugs.swift.org/browse/SR-8349
And a fix was merged, which should be available in Xcode 10.1 beta 2: https://github.com/apple/swift-xcode-playground-support/pull/29

**Comment by saoudrizwan** [comment]
## Fix
I just pushed a fix that replaces my `Swift.CustomStringConvertible` implementation using `JSONSerialization` with `String(describing: self.object as AnyObject)`. Everything should still work like before, except this new way has a much smaller memory footprint. I am able to use Playgrounds with DynamicJSON just fine now.

Hey @omidgolparvar can you please try out your code with the fix I just pushed, make sure to 
`pod 'DynamicJSON', '~> 1.0.3'`
